testApi/src/main/scala/africa/absa/testing/scapi/rest/response/action/types/AssertResponseActionType.scala

```diff
@@ -47,9 +47,21 @@ object AssertResponseActionType extends Enumeration {
   val CookieIsNotSecured: AssertResponseActionType.Value = Value("cookie-is-not-secured")
 
   // body-...
+  val BodyEquals: AssertResponseActionType.Value = Value("body-equals")
   val BodyContainsText: AssertResponseActionType.Value = Value("body-contains-text")
+  val BodyIsEmpty: AssertResponseActionType.Value = Value("body-is-empty")
+  val BodyIsNotEmpty: AssertResponseActionType.Value = Value("body-is-not-empty")
+  val BodyLengthEquals: AssertResponseActionType.Value = Value("body-length-equals")
+  val BodyStartsWith: AssertResponseActionType.Value = Value("body-starts-with")
+  val BodyEndsWith: AssertResponseActionType.Value = Value("body-ends-with")
+  val BodyMatchesRegex: AssertResponseActionType.Value = Value("body-matches-regex")
 
-  private val stringToValueMap = values.map(v => v.toString -> v).toMap
+  // body-json-...
+  val BodyJsonIsJsonArray: AssertResponseActionType.Value = Value("body-json-is-json-array")
+  val BodyJsonIsJsonObject: AssertResponseActionType.Value = Value("body-json-is-json-object")
+  val BodyJsonPathExists: AssertResponseActionType.Value = Value("body-json-path-exists")
 
+  // utils
+  private val stringToValueMap = values.map(v => v.toString -> v).toMap
   def fromString(s: String): Option[AssertResponseActionType] = stringToValueMap.get(s)
```

Comment:
```suggestion
def fromString(s: String): Option[AssertResponseActionType] = Try(this.withName(s)).toOption
```

Are you aware that Enums have built-in `withName` method that does what you need already? The only downside is that they throw `NoSuchElementException`s, but you could wrap it as outlined above.

Or you could just use `enum.withName` directly if that suited your use-case (you being ok with the exception).

Reply:
Fixed in commit bbe8482c8e309a0b48dda8c92c67cabd9b20e2a7.